src/pty.rs

```diff
@@ -0,0 +1,128 @@
+//! Create master and slave virtual pseudo-terminals (PTTYs)
+
+use std::ffi::CStr;
+use std::os::unix::io::RawFd;
+
+use libc;
+
+use {Error, fcntl, Result};
+
+/// Grant access to a slave pseudoterminal (see
+/// [grantpt(3)](http://man7.org/linux/man-pages/man3/grantpt.3.html))
+///
+/// `grantpt()` changezs the mode and owner of the slave pseudoterminal device corresponding to the
+/// master pseudoterminal referred to by `fd`. This is a necessary step towards opening the slave.
+#[inline]
+pub fn grantpt(fd: RawFd) -> Result<()> {
```

Comment:
Hmm, interesting. Would this be as simple as implementing that type and also `impl From<RawFd>` for it? We'd need to make sure that this can be easily used both with `std` and other `nix` functions. Additionally, is this sound in the context of #594?